nx/lib/nx/lin_alg/svd.ex

```diff
@@ -0,0 +1,168 @@
+defmodule Nx.LinAlg.SVD do
+  import Nx.Defn
+  @default_eps 1.1920929e-07
+
+  defn svd(input_tensor, opts \\ []) do
+    opts = keyword!(opts, max_iter: 100, eps: @default_eps)
+    {m, n} = Nx.shape(input_tensor)
+
+    tensor =
+      case Nx.shape(input_tensor) do
+        {m, n} when m < n ->
+          Nx.LinAlg.adjoint(input_tensor)
+
+        _ ->
+          input_tensor
+      end
+
+    {reduce_to_square, q, a} =
+      if m > 1.15 * n do
+        {q, a} = Nx.LinAlg.qr(tensor)
+        {true, q, a}
+      else
+        {false, tensor, tensor}
+      end
+
+    {u, s, v} = svd_tall_and_square(a, opts)
+
+    vt = Nx.LinAlg.adjoint(v)
+
+    case reduce_to_square do
```

Comment:
Still need to change the return so that we provide full matrices in the output to comply with current behaviour. Jax has the code for this, so it should be simple.